File: POO/TP1/pessoa.cpp
#include "pessoa.h"

int Pessoa::incrementador = 0;

Pessoa :: Pessoa(string nome){
    this-> nome = nome;     
    this-> id = ++incrementador;
}

Pessoa :: ~Pessoa(){};

string Pessoa :: getNome() const {
    return this->nome;
}

void Pessoa :: setNome(const string &nome){ 
    this->nome = nome; 
}



int Pessoa :: getID() const {
    return this->id; 
}   
void Pessoa :: setID(const int &id){
    this->id = id;
}



void Pessoa :: serialize(ostream & out) const{
    out << string("Nome: ") << getNome()
        << string("\nID: ") << getID() 
        << "\n\n";
}

ostream& operator <<(ostream& out, const Pessoa &obj){
    obj.serialize(out);
    return out;
}

File: POO/TP2/Trabalho-Pratico/src/Modelo/Pessoa.java
package Modelo;

import java.util.Objects;

public class Pessoa {
    private int id;
    private String nome;
    private String sobreNome;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSobreNome() {
        return sobreNome;
    }

    public void setSobreNome(String sobreNome) {
        this.sobreNome = sobreNome;
    }

    public Pessoa(int id, String nome, String sobreNome) {
        this.id = id;
        this.nome = nome;
        this.sobreNome = sobreNome;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Pessoa other = (Pessoa) obj;
        if (this.id != other.id) {
            return false;
        }
        if (!Objects.equals(this.nome, other.nome)) {
            return false;
        }
        return Objects.equals(this.sobreNome, other.sobreNome);
    }

    @Override
    public String toString() {
        return "Pessoa{" + "id=" + id + ", nome=" + nome + ", sobreNome=" + sobreNome + '}';
    }    
    
}